protos/flyteidl/event/event.proto

```diff
@@ -151,13 +151,54 @@ message TaskExecutionEvent {
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
 
+    // An optional explanation for the phase transition.
+    string reason                         = 13;
+
+
+    // A predefined yet extensible Task type identifier. If the task definition is already registered in flyte admin
+    // this type will be identical, but not all task executions necessarily use pre-registered definitions and this
+    // type is useful to render the task in the UI, filter task executions, etc.
+    string task_type                      = 14;
+
     // Metadata around how a task was executed.
     TaskExecutionMetadata metadata        = 16;
 }
 
+// This message contains metadata about external resources produced or used by a specific task execution.
+message ExternalResourceInfo {
+
+    // Identifiers for external resources created by this task execution, for example Qubole query ID or presto query ids.
+    repeated string external_ids = 1;
+}
+
+
+// This message holds task execution metadata specific to resource allocation used to manage concurrent
+// executions for a project namespace.
+message ResourcePoolInfo {
+    // Unique resource ID used to identify this execution when allocating a token.
+    string allocation_token               = 1;
+
+    // Namespace under which this task execution requested an allocation token.
+    string namespace                      = 2;
+}
+
 // Holds metadata around how a task was executed.
 // TODO(katrogan): Extend to include freeform fields (https://github.com/flyteorg/flyte/issues/325).
 message TaskExecutionMetadata {
+
```

Comment:
Can we remove the todo on the message? indicates task -
https://github.com/flyteorg/flyte/issues/325

Reply:
oh yes good catch, done